src/Workspaces/Core/Portable/Diagnostics/DiagnosticDataLocation.cs

```diff
@@ -13,15 +13,22 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     [DataContract]
     internal sealed class DiagnosticDataLocation
     {
+        /// <summary>
+        /// Path to where the diagnostic was originally reported.  May be a path to a document in a project, or the
+        /// project file itself.
+        /// </summary>
         [DataMember(Order = 0)]
-        public readonly DocumentId? DocumentId;
+        public readonly string OriginalFilePath;
```

Comment:
now required, and arguably the most important thing to provide.

Reply:
What happens if it's a project diagnostic and the project isn't backed by a file on disk?

Reply:
So, generally speaking, we'll use whatever the project itself states is its location.  Note that this is the data that ultimately ends up being shown to the user.  So this shoudl only be for real projects as those are what we produce diagnostics for.  If we had some sort of non-file-backed project that itself had diagnostics, it's unclear how those would ever surface in the first place.  HOwever, if they did, it would be against the path the project itself publishes :)

FilePath is a bit of a misnomer in this regard in that it's more like "pathy thing to show the user that is commonly a path that can be navigated to, but really just needs to be a string to show the user something sensible".